runtime/memprof.c

```diff
@@ -2281,6 +2287,42 @@ CAMLprim value caml_memprof_start(value lv, value szv, value tracker)
   CAMLreturn(config);
 }
 
+CAMLprim value caml_memprof_participate(value config)
+{
+  memprof_domain_t domain = Caml_state->memprof;
+  CAMLassert(domain);
+
+  if (Sampling(thread_config(domain->current))) {
+    caml_failwith("Gc.Memprof.participate: already profiling.");
+  }
+
+  switch (Status(config)) {
+  case CONFIG_STATUS_DISCARDED:
+    caml_failwith("Gc.Memprof.restart: profile already discarded.");
+  case CONFIG_STATUS_STOPPED:
+    caml_failwith("Gc.Memprof.restart: profile already stopped.");
+  case CONFIG_STATUS_SAMPLING:
+    break;
+  }
+
+  /* Orphan any surviving tracking entries from a previous profile. */
+  if (!orphans_create(domain)) {
+    caml_raise_out_of_memory();
+  }
+
+  set_config(domain, config);
```

Comment:
Just to check my understanding here: we always have full set of `memprof_thread_t` for every systhread of every domain, so the only thing we need to do to start tracing is to set their config.

Reply:
(and set the young_limit, of course).